ENIGMAsystem/SHELL/Universal_System/Instances/instance_create.h

```diff
@@ -106,6 +106,24 @@ namespace enigma_user
     return idn;
   }
 
+  enigma::instance_t instance_create_depth(int x, int y, int object, int depth) {
```

Comment:
I think the depth should be before the object. Check this [here](https://manual.yoyogames.com/#t=GameMaker_Language%2FGML_Reference%2FAsset_Management%2FInstances%2Finstance_create_depth.htm&rhsearch=instance_create_depth&rhhlterm=instance_create_depth) please.

Reply:
@RobertBColton Also why the implementation of functions is in the header file? I think we should create a CPP file for that ``instance_create.h`` file, what do u think?

Reply:
btw can't we just use this [function](https://github.com/enigma-dev/enigma-dev/blob/85c1a3ab9a2db140788ad6e744e24b7bbf194440/ENIGMAsystem/SHELL/Universal_System/Object_Tiers/object.h#L103) instead of just repeating code?

Reply:
Good catch
```suggestion
  enigma::instance_t instance_create_depth(int x, int y, int depth, int object) {
```

Reply:
The reason I did not use `object_set_depth` is because this function technically needs another optional paramter. We can't support the optional struct parameter yet to copy additional fields to initialize, but when we do it'll need to be in this file.

Reply:
cool then